nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -67,16 +68,22 @@ BtActionServer<ActionT>::BtActionServer(
   };
 
   if (!node->has_parameter("error_code_names")) {
-    std::string error_codes_str;
-    for (const auto & error_code : error_code_names) {
-      error_codes_str += error_code + "\n";
-    }
-    RCLCPP_WARN_STREAM(
-      logger_, "Error_code parameters were not set. Using default values of: "
-        << error_codes_str
-        << "Make sure these match your BT and there are not other sources of error codes you want "
-        "reported to your application");
+
+    const rclcpp::ParameterValue value = node->declare_parameter(
+      "error_code_names", 
+      rclcpp::PARAMETER_STRING_ARRAY);
+    if (value.get_type() == rclcpp::PARAMETER_NOT_SET) {
```

Comment:
Did you test this works? This isn't how I've seen it implemented before. It might work fine, but worth testing.

Reply:
Yep, 

I tested with no parameter and then set the param. Is there a better way of doing this? I couldn't find much.

Reply:
This works, typically I've just seen declaring and then using a `get` command to either have the return value be `false` that it couldn't be received (as the user shows below) or use the other `get` API which returns the value checking if it throws an exception.

I don't care one way or another though, if this works I'm perfectly happy with that